pandas/tests/series/test_operators.py

```diff
@@ -28,6 +28,38 @@
 from .common import TestData
 
 
+class TestDatetimeLikeArithmetic(object):
+    def test_sub_datetime64_not_ns(self):
+        # GH#7996
+        ser = Series(date_range('20130101', periods=3))
+        dt64 = np.datetime64('2013-01-01')
```

Comment:
test with a tz-aware as well (as I think that may hit your path, unless its getting caught earlier)

Reply:
It gets caught earlier: tz compat raises ValueError in ops._validate